teps/0138-decouple-api-and-feature-versioning.md

```diff
@@ -26,21 +39,21 @@ This document proposes updating Tekton Pipelines' feature flags design, as origi
 
-- [Feedback indicates](https://github.com/tektoncd/pipeline/issues/6592#issuecomment-1533268522) that users upgrading their CRDs from `v1beta1` to `v1` were confused to find `beta` features that worked by default in `v1beta1` did not work by default in `v1` when `enable-api-fields` was set to "`stable`" (its default value). This is especially confusing for users who are not cluster operators and cannot control the value of `enable-api-fields`, especially if they are not aware they are using `beta` features.
+- [Feedback indicates](https://github.com/tektoncd/pipeline/issues/6592#issuecomment-1533268522) that users upgrading their CRDs from v1beta1 to v1 were confused to find `beta` features that worked by default in v1beta1 did not work by default in v1 when `enable-api-fields` was set to "`stable`" (its default value). This is especially confusing for users who are not cluster operators and cannot control the value of `enable-api-fields`, especially if they are not aware they are using `beta` features.
 
-- For maintainers, the maintenance operation of swapping the storage version from `v1beta1` to `v1` should not have affected our users. However, we had to [change the user-facing default value of enable-api-fields from `stable` to `beta` ](https://github.com/tektoncd/pipeline/pull/6732) before changing the storage version of the API to [avoid breaking PipelineRuns using `beta` features](https://github.com/tektoncd/pipeline/pull/6444#issuecomment-1580926707).
+- For maintainers, the maintenance operation of swapping the storage version from v1beta1 to v1 should not have affected our users. However, we had to [change the user-facing default value of enable-api-fields from `stable` to `beta` ](https://github.com/tektoncd/pipeline/pull/6732) before changing the storage version of the API to [avoid breaking PipelineRuns using `beta` features](https://github.com/tektoncd/pipeline/pull/6444#issuecomment-1580926707).
 
 - When promoting features, it could cause confusions for contributors to be dependent on the fact whether an apiVersion is available. For example, during [the promotion to beta for projected workspaces](https://github.com/tektoncd/pipeline/pull/5530), the `v1` api's existence led to confusions of what to do with `beta` features in `v1beta1` and its difference with in `v1`.
 
-### Goals
+## Goals
 
 - Feature validations and implementation should be independant from any API version.
-- Come up with a backward-compatible migration plan for setting the `enable-api-fields` feature flag to `stable` in the long term.
+- Come up with a backward-compatible migration plan for setting the enable-api-fields feature flag to stable in the long term.
```

Comment:
The proposed solution changes enable-api-fields back to stable, but not in a backwards compatible way. Can this goal be reworded?

Reply:
I think I understand the point of switching back from "beta" to "stable" `enable-api-fields` is backward incompatible. But I might be wrong understanding the backwards incompatibility. Can I check if we provide with users the option to switch to another value so that they can continue to use the previous behaviour, can we say this solution is backward compatible?

(also related with the comment below)

Reply:
typically we use "backwards compatible" to mean that an api call that succeeded before the change will still succeed after the change (more details on how this applies in k8s here https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#on-compatibility). In this case, if you want to create a taskrun using beta features, it will succeed before the change but not after the change, unless the cluster operator modifies defaults. so I'd say this is not a backwards compatible change, but we can provide options to make the migration easier.

Reply:
Thanks for the clarifications! I updated it to providing an easier migration plan instead of it being backward compatible.